tests/phpunit/tests/speculative-loading/wpIsValidSpeculationRulesEagerness.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Tests for the wp_is_valid_speculation_rules_eagerness() function.
+ *
+ * @package WordPress
+ * @subpackage Speculative Loading
+ */
+
+/**
+ * @group speculative-loading
+ * @covers ::wp_is_valid_speculation_rules_eagerness
+ */
+class Tests_Speculative_Loading_wpIsValidSpeculationRulesEagerness extends WP_UnitTestCase {
+
+	/**
+	 * Tests that the function correctly identifies valid and invalid values.
+	 *
+	 * @ticket 62503
+	 * @dataProvider data_is_valid_speculation_rules_eagerness
+	 */
+	public function test_wp_is_valid_speculation_rules_eagerness( $eagerness, $expected ) {
+		if ( $expected ) {
+			$this->assertTrue( wp_is_valid_speculation_rules_eagerness( $eagerness ) );
+		} else {
+			$this->assertFalse( wp_is_valid_speculation_rules_eagerness( $eagerness ) );
+		}
```

Comment:
```suggestion
		$this->assertSame( $expected, wp_is_valid_speculation_rules_eagerness( $eagerness ) );
```

It will remove the conditional logic. Same apply to other tests 🔢

Reply:
I personally prefer going with the more explicit `assertTrue` and `assertFalse` that exist for this purpose. `assertSame` would of course work and avoid the conditional, but it would also alter the default messaging PHPUnit provides. Not a big deal, but either way it's probably a personal preference.